dimagi/commcare-hq
# Removed case claim session var config and fixed underlying bug for real

## Summary
https://dimagi-dev.atlassian.net/browse/QA-2427

This reverts https://github.com/dimagi/commcare-hq/pull/29004/ and fixes the underlying issue in a better way.

The problem is that case claim uses `case_id` as a kind of local variable and this breaks down when there's already a `case_id` in the session.

Typical parent/child selection requires a `parent_id` and then a `case_id` which is fine. But if the two menus are also parent/child menus, or if you're using the new select parent first workflow that lets you pick two cases of the same type, the form requires `case_id` and then `case_id_<case_type>`. Using case search for the second selection fails: you see the search screen, but when you click submit, instead of getting results, CC tries to re-select the first case.

I "fixed" this previously by allowing users to pick the name of the case claim variable, because I thought the issue was the case claim name had to match the form's datum name, but it does not. It just can't be the name of a variable that's already in the session. It's even fine to use the same id for all case claim variables. I don't fully understand why - I'd expect reusing `search_case_id` to not work twice in the same session - but it seems like the case claim var doesn't persist in the session.

## Feature Flag
Case search & claim

## Product Description
This removes the "session variable" option from case claim options, which was added 2 weeks ago. This configuration is no longer necessary.

## Safety Assurance


### Automated test coverage

Test coverage in PR.

### QA Plan

Asking QA and AEs to verify. I tested that the original bug isn't present on mobile, and the fix doesn't change mobile behavior, but will ask QA to verify this as well.

### Rollback instructions
